plugins/modules/one_host.py

```diff
@@ -152,16 +152,19 @@ def __init__(self):
     def allocate_host(self):
         """
         Creates a host entry in OpenNebula
+        self.one.host.allocate returns ID of a host
         Returns: True on success, fails otherwise.
 
         """
-        if not self.one.host.allocate(self.get_parameter('name'),
-                                      self.get_parameter('vmm_mad_name'),
-                                      self.get_parameter('im_mad_name'),
-                                      self.get_parameter('cluster_id')):
-            self.fail(msg="could not allocate host")
-        else:
+        try:
+            self.one.host.allocate(self.get_parameter('name'),
+                                   self.get_parameter('vmm_mad_name'),
+                                   self.get_parameter('im_mad_name'),
+                                   self.get_parameter('cluster_id'))
             self.result['changed'] = True
+        except Exception as e:
```

Comment:
Just to be on the safe side, have you checked whether the other `if` statements you are modifying do not fall under the same situation? 
Methods:
- `one.host.status()`
- `one.host.delete()`
- `one.host.update()`
- `one.host.addhost()`
Don't they also trigger exceptions when parameters are invalid?

Reply:
they are all almost the same in terms of error handling so yeah I'll change it
here are 2 examples for `one.host.status()`, 1 when running with no parameters, 2 when running with wrong parameter
![image](https://github.com/user-attachments/assets/9618ce09-c50f-48c8-9aee-93c878b99995)

Reply:
So, in that case, wouldn't it be good to add the ``try-except`` around them below as well?

Reply:
Could you please show which if statemets are wrong as well? From my perspective other if statements look right

Reply:
oops, I completely missed them, thanks!

Reply:
Beware that by using this, we are assuming that those calls will raise exceptions when failing (which makes sense, but not all developers write code that way). I would suggest you confirm that if you haven't yet.

Other than that, LGTM.

Reply:
yeah, it's too much IMO, no need to use it for every call :)
I simply do not use this module that much, so if one day I would need to use this module directly then might as well check if these are necessary